### PR TITLE
Handle grep's "not found" error in pgrep_maps()

### PR DESCRIPTION
**Pull request by Jongy**
## Description
Handle grep's "not found" error in `pgrep_maps()`. We haven't encountered it by now, because `pgrep_maps()` was run only on Pythons, and we always have Pythons running.

## Motivation and Context
Get rid of this error in the logs:
````
[2021-06-12 10:22:28,386] DEBUG: gprofiler.utils: (["grep -lP '(?:^.+/ruby[^/]*$)' /proc/*/maps"]) exit code: 1
[2021-06-12 10:22:28,386] ERROR: gprofiler: ruby profiling failed
Traceback (most recent call last):
  File "/app/gprofiler/main.py", line 271, in _snapshot
    process_perfs.update(future.result())
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 388, in __get_result
    raise self._exception
  File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/app/gprofiler/ruby.py", line 62, in snapshot
    processes_to_profile = _find_ruby_processes()
  File "/app/gprofiler/ruby.py", line 23, in _find_ruby_processes
    return pgrep_maps(r"(?:^.+/ruby[^/]*$)")
  File "/app/gprofiler/utils.py", line 185, in pgrep_maps
    assert result.returncode in (
AssertionError: unexpected 'grep' exit code: 1, stdout b'' stderr b''
```

## How Has This Been Tested?
1. Added a check in our tests for no `ERROR` logs.
2. Verified that the check actually catches the current error
3. Fixed the current error

## Checklist:
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have updated the relevant documentation.
- [x] I have added tests for new logic.

### CODE REVIEW

File: gprofiler/utils.py [hunk 1]
```diff
@@ -181,12 +181,23 @@ def pgrep_maps(match: str) -> List[Process]:
...
+        1,
         2,
     ), f"unexpected 'grep' exit code: {result.returncode}, stdout {result.stdout!r} stderr {result.stderr!r}"
 
+    error_lines = []
+    for line in result.stderr.splitlines():
+        if not (line.startswith(b"grep: /proc/") and line.endswith(b"/maps: No such file or directory")):
+            error_lines.append(line)
+    if error_lines:
+        logger.error(f"Unexpected 'grep' error output (first 10 lines): {error_lines[:10]}")
```
Comment (michelhe) [inline gprofiler/utils.py:199]: I think thats overdoing it, no ?

Comment (Jongy) [reply gprofiler/utils.py:199]: I don't want to ignore errors

**Review by michelhe — APPROVED** [review]
LGTM

**Comment by Jongy** [comment]
```
[2021-06-13 08:59:49,124] ERROR: gprofiler: java profiling failed
Traceback (most recent call last):
  File "psutil/_pslinux.py", line 1576, in wrapper
  File "psutil/_pslinux.py", line 1697, in cmdline
ProcessLookupError: [Errno 3] No such process

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "gprofiler/main.py", line 271, in _snapshot
  File "concurrent/futures/_base.py", line 425, in result
  File "concurrent/futures/_base.py", line 384, in __get_result
  File "concurrent/futures/thread.py", line 56, in run
  File "gprofiler/java.py", line 342, in snapshot
  File "gprofiler/utils.py", line 171, in <genexpr>
  File "psutil/__init__.py", line 669, in exe
  File "psutil/__init__.py", line 644, in guess_it
  File "psutil/__init__.py", line 677, in cmdline
  File "psutil/_pslinux.py", line 1580, in wrapper
psutil.NoSuchProcess: psutil.NoSuchProcess process no longer exists (pid=4830)
```

Weird errors

**Review by michelhe — APPROVED** [review]

**Pull request merged**